### PR TITLE
Anderson-Darling test

### PR DESCRIPTION
**Pull request by wildart**
Anderson-Darling test
- [x] one-sample test
- [x] k-sample test
- [x] documentation

Links:
- http://www.itl.nist.gov/div898/handbook/eda/section3/eda35e.htm
- http://links.jstor.org/sici?sici=0162-1459%28198709%2982%3A399%3C918%3AKAT%3E2.0.CO%3B2-9

### CODE REVIEW

**Comment by wildart** [comment]
Ready for review.

File: test/runtests.jl [hunk 1]
```diff
@@ -8,3 +8,6 @@ include("mann_whitney.jl")
 include("t.jl")
 include("wilcoxon.jl")
 include("power_divergence.jl")
+include("anderson_darling.jl")
+include("anderson_darling.jl")
```
Comment (andreasnoack) [inline test/runtests.jl]: One too many here

File: src/anderson_darling.jl [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    # q = [.05, .10, .15, .20, .25, .30, .35, .40, .45, .50, .55, .60, .65, .70, .75, .80, .85, .90, .95, .975, .99, .995]
+    # b0 = [-.512, -.552, -.608, -.643, -.707, -.735, -.772, -.770, -.778, -.779, -.803, -.818, -.818, -.801, -.800, -.756, -.749, -.750, -.795, -.881, -1.013, -1.063]
+    # b1 =[2.10, 1.25, 1.07, .93, 1.03, 1.02, 1.04, .90, .80, .67, .70, .58, .42, .12, -.09, -.39, -.59, -.8, -.89, -.94, -.93, -1.34]
+    # a₊ = [.1674, .1938, .2147, .2333, .2509, .2681, .2853, .3030, .3213, .3405, .3612, .3836, .4085, .4367, .4695, .5091, .5597, .6305, .7514, .8728, 1.0348, 1.1578]
+    # a = a₊.*(1.+ b0/x.n +b1/x.n/x.n)
+    # qi = searchsortedlast(a, x.A²)
+    # Z(i) = a₊[i]*(1+b0[i]/x.n +b1[i]/x.n/x.n)
+
+    Z = [.200, .340, .600]
```
Comment (andreasnoack) [inline src/anderson_darling.jl]: Maybe avoid the array construction and extraction of values here and just substitute in the values.

**Comment by andreasnoack** [comment]
I'll try to review a little more carefully tomorrow.

File: src/anderson_darling.jl [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    k < 2 && error("Need at least two samples")
+
+    n = map(length, samples)
+    Z = sort(vcat(samples...))
+    N = length(Z)
+    Z⁺ = unique(Z)
+    L = length(Z⁺)
+
+    L < 2 && error("Need more then 1 observation")
+    any(map(l->l == 0, n)) && error("One of the samples is empty")
```
Comment (andreasnoack) [inline src/anderson_darling.jl]: Maybe `minimum(n)==0` is a little less convoluted.

**Comment by andreasnoack** [comment]
I think this looks good. Great documentation and tests.

**Comment by andreasnoack** [comment]
@wildart Is this ready now?

**Comment by wildart** [comment]
Yes

**Comment by andreasnoack** [comment]
Great.

**Pull request merged**